lib/common/utils.c

```diff
@@ -69,121 +69,95 @@ static const char *crm_app_description = NULL;
 static char *crm_short_options = NULL;
 static const char *crm_app_usage = NULL;
 
-gboolean
-check_time(const char *value)
+bool
+pcmk__valid_time(const char *value)
 {
-    if (crm_get_msec(value) < 5000) {
-        return FALSE;
-    }
-    return TRUE;
+    return crm_get_msec(value) >= 5000;
 }
 
-gboolean
-check_timer(const char *value)
+bool
+pcmk__valid_timer(const char *value)
 {
-    if (crm_get_msec(value) < 0) {
-        return FALSE;
-    }
-    return TRUE;
+    return crm_get_msec(value) >= 0;
 }
 
-gboolean
-check_boolean(const char *value)
+bool
+pcmk__valid_boolean(const char *value)
 {
-    int tmp = FALSE;
+    int tmp;
 
-    if (crm_str_to_boolean(value, &tmp) != 1) {
-        return FALSE;
-    }
-    return TRUE;
+    return crm_str_to_boolean(value, &tmp) == 1;
```

Comment:
crm_str_to_boolean is a pretty weird function.  First, it doesn't actually convert to a boolean.  It converts to an int that just happens to overlap with the bool type.  Second, why couldn't its return type also be a boolean indicating whether it succeeded or not?  Then this could be reduced to `return crm_str_to_boolean(value, &tmp);`.

Reply:
Agreed, but it's public API :-/